### PR TITLE
[SYCL][Graph] Clean-up E2E Tests

### PR DESCRIPTION
**Pull request by EwanC**
Some clean-up for SYCL-Graph E2E tests:
* Remove redundant `Event` variables that are initialized over loop iterations but never used.
* Remove all instances of the no immediate command-list property, and use environment variable instead to test both paths.
* Always use FileCheck leak checking rather than `CHECK-NOT: Leak`.
* Remove unnecessary threading code from `Inputs/basic_usm.cpp`

### CODE REVIEW

**Comment by mfrancepillois** [comment]
PR https://github.com/intel/llvm/pull/12599 (opened 2 weeks ago) touches on the same files. Perhaps we can wait until PR #12599 is merged before going any further on this PR?

File: sycl/test-e2e/Graph/Inputs/basic_usm.cpp
Comment (mfrancepillois) [inline sycl/test-e2e/Graph/Inputs/basic_usm.cpp:43]: Can be removed. It is not used in the test.

**Review by mfrancepillois — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers Can we merge this please

**Pull request merged**
